easybuild/easyblocks/o/optislang.py

```diff
@@ -0,0 +1,107 @@
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing ANSYS optiSLang, implemented as an easyblock
+
+@author: Chia-Jung Hsu (Chalmers University for Technology)
+"""
+import os
+import re
+import stat
+import tempfile
+
+from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions
+from easybuild.tools.run import run_cmd
+
+
+class EB_optiSLang(PackedBinary):
+    """Support for installing ANSYS optiSLang."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize optiSLang-specific variables."""
+        super(EB_optiSLang, self).__init__(*args, **kwargs)
+        self.ansysver = None
+
+    def install_step(self):
+        """Custom install procedure for ANSYS optiSLang."""
+        licservs = self.cfg['license_server']
+        if licservs is None:
+            licservs = os.getenv('EB_ANSYS_LICENSE_SERVER', 'license.example.com')
+        licservs = licservs.split(',')
+        licport = self.cfg['license_server_port']
+        if licport is None:
+            licport = os.getenv('EB_ANSYS_LICENSE_SERVER_PORT', '2325:1055')
+        licopts = ['-licserverinfo %s:%s' % (licport, serv) for serv in licservs]
+        licoptsstr = ' '.join(licopts)
+
+        tmpdir = tempfile.mkdtemp()
+
+        # Sources (e.g. iso files) may drop the execute permissions
+        adjust_permissions('INSTALL', stat.S_IXUSR)
+        cmd = "./INSTALL -silent -install_dir %s -usetempdir %s %s" % (self.installdir, tmpdir, licoptsstr)
+        run_cmd(cmd, log_all=True, simple=True)
+
+        adjust_permissions(self.installdir, stat.S_IWOTH, add=False)
+
+    def set_ansysver(self):
+        """Determine name of version subdirectory in installation directory."""
+        entries = os.listdir(self.installdir)
+        version_dir_regex = re.compile('^v[0-9]+')
+        version_dirs = [e for e in entries if version_dir_regex.match(e)]
+        if len(version_dirs) == 1:
+            self.ansysver = version_dirs[0]
+        else:
+            raise EasyBuildError("Failed to isolate version subdirectory in %s: %s", self.installdir, entries)
+
+    def make_module_req_guess(self):
+        """Custom extra module file entries for ANSYS optiSLang."""
+
+        if self.ansysver is None:
+            self.set_ansysver()
+
+        guesses = super(EB_optiSLang, self).make_module_req_guess()
+        dirs = [
+            'aisol/bin/linx64',
+            'dpf/bin/linux64',
+            'optiSLang',
+        ]
```

Comment:
CI tests are broken because they assume you can always do a --module-only even on an empty dir. 

Perhaps we can change this to just glob for these paths instead, and just add those (and if the directory is empty it just adds nothing).

Reply:
Done